sdk/core/azure-core/test/ut/transport_adapter_curl.cpp

```diff
@@ -37,9 +37,9 @@ namespace Azure { namespace Core { namespace Test {
   } // namespace
 
 /***********************  Unique Tests for Libcurl   ********************************/
-//  Disabling test with TEST_DISABLED. The test name cannot be changed because it depends on a
-//  friend class definition. Hence, it can't use the gtest DISABLE_
-#if !defined(TEST_DISABLED)
+//  Disabling test with INCLUDE_DISABLED_TESTS. The test name cannot be changed because it depends
+//  on a friend class definition. Hence, it can't use the gtest DISABLE_
+#if defined(INCLUDE_DISABLED_TESTS)
```

Comment:
Is using `defined` necessary here?

It's fine to do, just curious, since the guidelines have just `#if INCLUDE_DISABLED_TESTS` as examples.

Reply:
https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-use-hashif is happy with `#if defined(MEOW)`. 

"Always use `#if` if you have to use the preprocessor. This works fine, and does the right thing, even if `DEBUG` is not defined at all (!)" seems oblivious to the fact that most compilers have a warning that fires when an undefined identifier appears in a preprocessor conditional. For that reason I prefer to use definition testing for external configuration - things that users define, since they don't always define them - and value testing for internal configuration that we can ensure is always defined even if the user doesn't override the value.

Reply:
Can you please clarify what you mean by "definition testing" and "value testing"?

> seems oblivious to the fact that most compilers have a warning that fires when an undefined identifier appears in a preprocessor conditional.

Do we need a guidelines update, then?

Reply:
@ahsonkhan  the guidelines tell us to use `define`:
```
If you really need to test whether a symbol is defined or not, test it with the defined() construct, which allows you to add more things later to the conditional without editing text that’s already in the program:
```

![image](https://user-images.githubusercontent.com/24213737/101106116-3ff66680-3584-11eb-85ee-9aab19912b3d.png)

Reply:
I already read that part @vhvb1989, but right before then, it says `#if DEBUG` is a "// Good example". So, some examples seem to indicate that using `#if` for preprocessor values are known/expected to not be defined, is fine.

It looks to me, we'd use `#if defined(...)` in cases where we might actually define that named preprocessor, have it documented, or if the customer would define it.

Same with this:
![image](https://user-images.githubusercontent.com/6527137/101106556-299cda80-3585-11eb-800a-52063184a38c.png)

Reply:
@ahsonkhan, the `#if` alone check for a boolean condition. No matter what comes after it.
If we add a macro like `#if SYMBOL`, the if will work only if SYMBOL represents a boolean expression, but will fail (compile error if not).
```cpp
// This example won't compile
#define SYMBOL "string macro"

#if SYMBOL
 std::cout << "hello";
#endif
```

Then, using `#if` alone will require us to define the symbol always with a boolean value.

```cpp
// This example won't compile
// Just define the symbol without any value
#define SYMBOL

#if SYMBOL
 std::cout << "hello";
#endif
```

That's why we use `define()`. We don't care about the value of the MACRO, we only care if it is defined or not